fizzbuzz/src/main/java/fizz.buzz/number/factory/NumberFactoryImpl.java

```diff
@@ -1,23 +1,29 @@
 package fizz.buzz.number.factory;
 
-import fizz.buzz.number.*;
 import fizz.buzz.number.Number;
+import fizz.buzz.number.NumberFactory;
 
 public class NumberFactoryImpl implements NumberFactory {
-    public Number build(Integer input) {
 
-        if (input % 3 == 0 && input % 5 == 0) {
-            return new FizzBuzzNumber(input);
-        }
+    private FizzBuzzNumberFactory numberFactory;
+
+    public NumberFactoryImpl() {
+        numberFactory = new FizzBuzzNumberFactory();
+
+        setupProcessingChain();
+    }
 
-        if (input % 3 == 0) {
-            return new FizzNumber(input);
-        }
+    private void setupProcessingChain() {
```

Comment:
I think the order of the processing chain is very important.
Having numberFactory already instantiated in the constructor seems to be "too far away". I think it should also go in this method.

I would also recommend to order the setSuccessor calls accordingly to improve the readability.

```
        RegularNumberFactory regularFactory = new RegularNumberFactory();
        BuzzNumberFactory buzzFactory = new BuzzNumberFactory();
        FizzNumberFactory fizzFactory = new FizzNumberFactory();
        FizzBuzzNumberFactory fizzBuzzFactory = new FizzBuzzNumberFactory();

        fizzBuzzFactory.setSuccessor(fizzFactory);
        fizzFactory.setSuccessor(buzzFactory);
        buzzFactory.setSuccessor(regularFactory);

        numberFactory = fizzBuzzFactory;

```